### ISSUE TITLE
Graphql: When specifying a Neo field twice in a Query, nested children are returned at the top-level

### ISSUE THREAD

**Issue by drobin03** [spicywebau/craft-neo#425]
#### Description

When querying content of a Neo field that exists on multiple entry types, as soon as a second reference to the field is added to the query (for the second entry type), all nested children are duplicated and returned at the field-level as well as being returned in the children array.

#### Steps to reproduce

Our use-case is more complex, but we managed to boil it down to the following conditions to reproduce the issue: 

1. Create a simple Neo field `body`, with a single content block type `hero` and a child block `heading`
2. Configure two entry types (`page`, `article`), add the `body` field to both
3. Create a page with the slug `test`
4. Populate a `hero` with a single `heading` on the `body` field
5. **Run the following query in graphiql**

```graphql
query {
  entry(uri: ["test"]) {
    __typename
    ... on pages_page_Entry { # entry type
      body { # Neo Field
        __typename
      }
    }
    ... on articles_article_Entry { # entry type
      body { # Neo Field
        __typename
      }
    }
  }
}
```

##### Expected Result
```graphql
{
  "data": {
    "entry": {
      "__typename": "pages_page_Entry",
      "body": [
        {
          "__typename": "body_hero_BlockType",
          "children": [
            {
              "__typename": "body_heading_BlockType"
            },
          ]
        },
      ]
    }
  }
}
```

##### Actual Result
```graphql
{
  "data": {
    "entry": {
      "__typename": "pages_page_Entry",
      "body": [
        {
          "__typename": "body_hero_BlockType",
          "children": [
            {
              "__typename": "body_heading_BlockType"
            },
          ]
        },
        #### --- 
        { 
          "__typename": "body_heading_BlockType"
        },
        #### --- This shouldn't be here
      ]
    }
  }
}
```

**Note the extra `body_heading_BlockType` printed at the top level**

We were also able to determine that we could get the correct response in a few ways:

1. Renaming the `body` field on the second reference
```graphql
query {
  entry(uri: ["test"]) {
    __typename
    ... on pages_page_Entry {
      body {
        __typename
      }
    }
    ... on articles_article_Entry { 
      articleBody: body {
        __typename
      }
    }
  }
}
```
2. Changing the order, the last one specified seems to work properly
```graphql
query {
  entry(uri: ["test"]) {
    __typename
    ... on articles_article_Entry { 
      body {
        __typename
      }
    }
    ... on pages_page_Entry {
      body {
        __typename
      }
    }
  }
}
```
3. Removing the second reference
```graphql
query {
  entry(uri: ["test"]) {
    __typename
    ... on pages_page_Entry {
      body {
        __typename
      }
    }
  }
}
```

#### Other information
- Neo version: 2.8.16
- Craft version: 3.5.17.1
- (if multi-site) What is the affected Neo field's propagation method: n/a
- Is eager-loading used? No

**Comment by ttempleton** [comment]
Thanks for reporting this, @drobin03, confirming that I can reproduce it.  Will look into it further and hopefully have a fix released soon.

**Comment by drobin03** [comment]
Fantastic, thanks @ttempleton!

**Comment by ttempleton** [comment]
Should now be fixed in Neo 2.8.17!
